device-connectors/src/testflinger_device_connectors/devices/noble_oemscript/noble_oemscript.py

```diff
@@ -0,0 +1,28 @@
+# Copyright (C) 2023 Canonical
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Ubuntu OEM Script provisioning for OEM devices with Ubuntu Noble series
+For systems that use the oem image-deploy.sh script for provisioning
+"""
+
+import logging
+from testflinger_device_connectors.devices.oemscript.oemscript import OemScript
+
+logger = logging.getLogger(__name__)
+
+
+class NobleOemScript(OemScript):
+    """Device Agent for Noble OEM devices."""
+
+    distro = "noble"
```

Comment:
This looks like a much nicer process that will cover more than just noble in the future, and there's probably a better name for it than noble_oemscript. It should also be possible to install the next LTS with this too, right? It seems like this should be a new device connector and just copy the parts of OemScript that make sense, modify it, and give it a new name. I'm not sure what to call it though - maybe you have an internal name for what you call this new version of oemscript recovery?

However, if this is ONLY ever expected to work with noble, then perhaps "noble_oemscript" isn't such a bad name, but that would be kind of sad I think.

Reply:
Agree. I renamed it to oem_autoinstall since this is used instead of ubuntu recovery now. I hope the next LTS will be using the same process.